llvm/test/Transforms/ConstraintElimination/transfer-samesign-facts.ll

```diff
@@ -303,3 +303,23 @@ define i1 @ugt_assumed_positive_values(i8 %a, i8 %b) {
 
   ret i1 %result
 }
+
+define i1 @samesign_flipped_signedness(i8 %a)  {
```

Comment:
Can you add the test https://github.com/llvm/llvm-project/issues/120089#issuecomment-2635581659?